source/sonata_report.rst

```diff
@@ -86,6 +88,76 @@ For a full compartment reports, you need to specify both ``"sections": "all"`` a
+    *   ``node_id``: The ID of the node within the specified ``population``.
+    *   ``section_name``: The name of the  NEURON section. Currently, only ``"soma"``, ``"dend"``, ``"apic"`` and ``"axon"`` are supported. This should not contain the ``section_index`` e.g. ``"dend[10]"``.
+    *   ``section_index``: The number of the NEURON section index e.g. for dend[10], ``section_index`` is 10. Smallest ``section_index`` is 0 in NEURON.
+    *   ``location``: The fractional distance along the section (0<= location <=1).
+
+
+Simulation Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+To use compartment sets, first declare the ``compartment_sets.json`` at the top level of your ``simulation_config.json``. Then, in your compartment report definition, set ``"sections": "compartment_set"`` and use the ``"compartments"`` field to specify the name of the desired set from your ``compartment_sets.json`` file.
+
+.. code-block:: json
+
+  {
+    "compartment_sets_file": "circuit/compartment_sets.json",  // Path to your compartment sets file
+
+    "reports": {
+      "dend_report_v": {                            // Name of the report
+        "type": "compartment",
+        "sections": "compartment_set",              // Use "compartment_set"
+        "compartments": "example_compartment_set",  // Name of the set from compartment_sets.json
+        "variable_name": "v",
+        "unit": "mV",
+        "dt": 0.1,
+        "start_time": 0.0,
+        "end_time": 100.0
+      }
+      // ... other reports ...
+    }
+    // ... other global configurations ...
+  }
+
+Key changes:
+
+*   **``sections``**:
+    *   When set to ``"compartment_set"``, it indicates that the report targets are defined by a named set in the ``compartment_sets.json`` file.
+    *   Previously supported values for ``sections`` include ``"soma"``, ``"axon"``, ``"dend"``, ``"apic"``, or ``"all"``. These continue to function as before, typically used with the ``cells`` key to specify target populations.
+*   **``compartments``**:
+    *   If ``sections`` is ``"compartment_set"``, this field **must** contain the name of a key (a specific compartment set) defined in your ``compartment_sets.json`` file (e.g., ``"example_compartment_set"``).
+    *   For other ``sections`` types (``"soma"``, ``"axon"``, ``"dend"``, ``"apic"``, or ``"all"``), ``compartments`` typically takes values like ``"center"`` or ``"all"``.
+*   **``cells``**: The ``cells`` key (e.g., ``"cells": "Mosaic"`` or ``"cells": ["popA", 123]``) is **not allowed** and should cause an error if ``sections`` is ``"compartment_set"``. The selection of cells and their specific compartments is entirely managed by the chosen compartment set from ``compartment_sets.json``.
+
+The output HDF5 report format for these targeted compartment reports remains the same as described in the main :ref:`compartment_report_main` section.
+The ordering of axon/apic/dend/soma and the gid, type pair in report can only show up once.
```

Comment:
to me this is not clear. `[0, 'dend', ..., ...]` can only appear once? The ordering can only appear once?

Reply:
> Also add 2 examples, what is not possible and what is possible

Added one example with acceptable entries in `compartment_set` as mentioned in the last comment. Also, added another one for raising a warning for duplicate entries.